crates/client/src/fuzzer/snapshot_generator.rs

```diff
@@ -256,9 +278,15 @@ fn create_snapshot_struct(
                             (true, true) => {
                                 Ok(quote! {pub #field_name: Option<&'info #field_type>,})
                             }
-                            (true, _) => Ok(quote! {pub #field_name: Option<#field_type>,}),
+                            (true, _) => {
+                                let field_type = construct_full_path(&field_type.to_token_stream(),parse_result,name).unwrap_or(field_type.clone());
```

Comment:
Arguments passed to unwrap_or are eagerly evaluated; if you are passing the result of a function call, it is recommended to use [unwrap_or_else], which is lazily evaluated.